### DISCUSSION TITLE
Adding options for `embed/chat/readonly/`?

### DISCUSSION THREAD

**Discussion by DidierMalenfant** [owncast/owncast#3464] [Ideas] [answered]
I use `embed/chat/readonly/` to embed the chat into my OBS stream.

Would it be possible to add an option to disable displaying the chat welcome message and also disable the `stream about to start` and `stream about to end` messages for this endpoint?

I don't know enough about CSS but maybe this can already be done easily by setting the colors for these messages as transparent? Would be an easy stop-gap solution without polluting the code with more options.

**Comment by gabek** [comment]
If you remove your welcome message in the chat settings, it shouldn't send anything. So that's an easy change.

As for removing specific messages, you're right, CSS would be the solution to that. You wouldn't need to change the color, but if you target the messages, you can probably specify to hide them. Though I've never tried it myself, I don't see why it wouldn't work.

```css
display: none;
```

**Reply by DidierMalenfant** [reply]
Thanks I'll give the CSS route a try. That will help me up my game in that area.

Re: the welcome message, I actually DO want it for the regular chat users, just not for the readonly chat in OBS.

**Comment by gabek** [comment]
Ohhh. I'm sure you could accomplish that with CSS as well, but it will be more complex. You might have to target the actual text in the message, such as the name or something. And you wouldn't be able to use the internal Owncast CSS support for that, but you'd have to use the OBS functionality where it can inject specific CSS.

**Comment by DidierMalenfant** [comment, marked as answer]
Ok I now know way more CSS than I ever wanted to 😂

Here is something that can be added to the `Custom CSS` setting in OBS to remove welcome messages in the chat:
```
[class^="ChatSystemMessage"] {
   display: none;
}
```
and similarly for the `Stream Starting`/`Stream Ending` messages:
```
[class^="ChatActionMessage"] {
   display: none;
}
```
